### PR TITLE
Feature/time input

### LINKED ISSUE #30: React Native Date Picker (Timepicker)
~~[React Native Date Picker](https://www.npmjs.com/package/react-native-date-picker)~~ <= Doesn't work with Expo :(
[DateTimePicker](https://github.com/react-native-datetimepicker/datetimepicker)

### PR DESCRIPTION
**Pull request by holladayian**
### Participating Group Members:
Ian
 
### Code Highlights:
AddTask is getting kind of hairy, and honestly could go for a refactor, but I think we can save that for an extension/if they let us work on that stuff after code freeze.
 
### Have Tests Been Added?
- [x] No.
- [ ] Yes, but all tests are not passing. 
- [ ] Yes, and all are passing.
 
### Any background context you want to provide?
Lost added to state. I know I keep seeing React-Native used with hooks. But once again, that can be an extension.
 
### Any packages installed?
DateTimePicker (from the issue)
 
### Message/Questions for reviewer:
Does everything work they way you'd like it? Can you break it? (It would make me cry)
 
### Issues: 
 Closes #30 

### Screenshots (if appropriate): 
https://user-images.githubusercontent.com/62005294/104111894-5df66b80-5294-11eb-82a1-3af9d9be095f.mov

 
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] All new and existing tests passed  
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
 
## Tags (for mentor/alumni code review)
@D-Lessenden 
@nicorithner 
@sarynm12

### CODE REVIEW

File: components/AddTask.js [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 import { Button } from "react-native-elements";
 
 class AddTask extends React.Component {
   constructor() {
     super();
     this.state = {
       name: '',
-      time: '',
-      window: '',
+      time: new Date(),
```
Comment (kathrynljackson) [inline components/AddTask.js:11]: Does this set `this.state.date` to the current time/date?

File: package.json [hunk 2]
```diff
@@ -9,6 +10,7 @@
     "react": "~16.13.0",
     "react-dom": "~16.13.0",
     "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
+    "react-native-date-picker": "^3.2.7",
```
Comment (kathrynljackson) [inline package.json:13]: We will need to make sure we run `npm install`!

File: components/AddTask.js [hunk 3]
```diff
@@ -32,32 +31,57 @@ class AddTask extends React.Component {
...
+        <TextInput placeholder={this.state.placeholderTime} style={styles.input} onFocus={() => this.setState({ viewTime: true }) }>
+          {this.state.viewTime && 
+            <Text>{this.state.time.getHours().toString()}:{this.state.time.getMinutes().toString()}</Text>}
+        </TextInput>
+        <TextInput placeholder={this.state.placeholderWindow} style={styles.input} onFocus={() => this.setState({ viewWindow: true }) }>
+          {this.state.viewWindow && 
+            <Text>{this.state.window.getHours().toString()}:{this.state.window.getMinutes().toString()}</Text>}
+        </TextInput>
+        {this.state.viewTime && 
+          <DateTimePicker value={this.state.time} display='spinner' onChange={this.onChangeTime} is24Hour={true} mode={'time'}/>}
```
Comment (kathrynljackson) [inline components/AddTask.js:82]: I'm glad the `spinner` worked`. It looks great!

**Review by kathrynljackson — APPROVED** [review]
This looks great! Easy to use and understand without any directions.

File: components/AddTask.js (hunk 1 above)
Comment (holladayian) [reply components/AddTask.js:11]: I believe so. And now that you mention it it's not quite the correct syntax. I needed this to start out as a date for the DateTimePicker to work, but they're only conditionally rendered based onFocus of their respective TextInput. This should be okay to continue, but I will make an issue to refactor.

File: components/Caretaker.js [hunk 4]
```diff
@@ -6,9 +6,9 @@ import AddTask from './AddTask';
 
 const Caretaker = ({history, navigation}) => {
   const [timers, editTimers] = useState([
-    {id: 1, name: "Wake-up", time: '7:00', window: '0:15'},
-    {id: 2, name: "Take meds", time: '12:00', window: '1:00'},
-    {id: 3, name: "go to sleep", time: '22:00', window: '1:00'}
+    {id: 1, name: "Wake-up", time: new Date(), window: new Date()},
+    {id: 2, name: "Take meds", time: new Date(), window: new Date()},
+    {id: 3, name: "go to sleep", time: new Date(), window: new Date()}
```
Comment (nware1066) [inline components/Caretaker.js:11]: I think this is better than a typed in user input, so we can control the type

**Review by nware1066 — APPROVED** [review]

**Pull request merged**